sdks/python/apache_beam/io/gcp/pubsub.py

```diff
@@ -127,6 +130,10 @@ def _from_message(msg):
     """
     # Convert ScalarMapContainer to dict.
     attributes = dict((key, msg.attributes[key]) for key in msg.attributes)
+    # Parse the PubSub message_id and add to attributes
+    sysattribs = dict(message_id=msg.message_id)
```

Comment:
This would override a user-defined 'message_id' attribute. Better to make message_id a member of PubsubMessage.

Reply:
Yes I agree.  This was merely done to fit with the existing documentation, however I feel that making the PubSubMessage mimic the protobuf definition makes more sense (and can add the publish_time attribute as well).  In addition, due to the handling of pubsub in dataflow, the message_id is currently not exposed and so is simply an empty attribute at that point. I'm thinking to close this pull request and update the issue with a dependency on the dataflow worker being amended (unfortunately I'm even worse at java than python!)

Reply:
I investigated the issue yesterday and found the issue was with the Windmill service leaving out message id and publish_time. We can go ahead with this PR, and leave a note saying message_ids will work once Windmill service is updated (point to BEAM-7819 for details).

FYI, I believe PR #8370 will also require the Windmill update for message ids to work on Dataflow (but I'm not sure when Java SDK uses Windmill).

Reply:
I've started to look at moving message_id out of the attributes and into it's own attribute, along with publish_time. The latter causes some issues with testing, as it cannot be updated in the protobuf, so the way in which some of the mocked up tests work doesn't quite work.

As such, I'm going to focus on message_id first, & have a think how to deal with the publish_time later.